lxd/storage/backend_lxd.go

```diff
@@ -5590,6 +5590,43 @@ func (b *lxdBackend) UpdateCustomVolume(projectName string, volName string, newD
 			}
 		}
 
+		sharedVolume, ok := changedConfig["security.shared"]
```

Comment:
This check is only relevant to block volumes, but we appear to be doing it for all custom volume types.

Reply:
since the `security.shared` config key is only allowed to be set on block volumes, it will always be unset to all other types and thus the checks won't be done for them. That said, I could add an explicit check here if you prefer, or add a comment to make this behavior clearer.

Reply:
Yes please - otherwise this code can be brittle and non-obvious if it depends on external scenarios be fulfilled.